### PR TITLE
Add Hacker News Challenge

### PR DESCRIPTION
**Pull request by raven300**
Signed-off-by: radha <rads_venkatesan@yahoo.com>

### CODE REVIEW

File: COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md [hunk 1]
```diff
@@ -203,3 +203,55 @@ I hope that by taking this challenge, you will be able to learn something new!
...
+
+\newpage
+## Hacker News Rankings for Mainframe/COBOL Posts
+
+![](Images/hacker-img1.png)
+
+We will explore the popular Hacker News website for this challenge. Hacker News is an online community started by Paul Graham for sharing "Anything that good hackers would find interesting. That includes more than hacking and startups".
+
+
+### A little Background
```
Comment (zeibura) [inline COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md:215]: fix capitalisation - A Little Background

File: COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md [hunk 2]
```diff
@@ -203,3 +203,55 @@ I hope that by taking this challenge, you will be able to learn something new!
...
+
+### The plan
+ [] There are different creative ways of accomplishing this but here’s our plan: We will have a COBOL program that reads the input CSV file and retrieves only the ***Mainframe/COBOL*** stories. It then calculates the ranking score for the stories by factoring in the time they were posted and the number of votes they received. Each of the records is then written to an output dataset along with the ranking score. 
+
+ [] We will then use `DFSORT` to sort the output dataset on ranking score, highest to lowest and display the posts as a simple report mimicking the front page. 
+ 
+ Let's get started!
+1. Take a look and familialize yourself with the dataset on z/OS: `ZOS.PUBLIC.HACKER.NEWS`. This is a CSV file that serves as input to your COBOL program. The file was created by downloading [this Kaggle dataset](https://www.kaggle.com/hacker-news/hacker-news-posts), removing the lengthy `URL` column that is of no relevance to us and uploading it to z/OS. You can directly reference this DS in your JCL. Please avoid making a copy as it is fairly large with around 300,000 records.
```
Comment (zeibura) [inline COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md:227]: familiarize

File: COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md [hunk 3]
```diff
@@ -203,3 +203,55 @@ I hope that by taking this challenge, you will be able to learn something new!
...
+
+### Our Goal
+We will be working on a Hacker News 2015-2016 dataset from Kaggle with a full year’s worth of stories:  Our goal is to extract only the Mainframe/COBOL related stories and assign ranking scores to them based on (a simplified version) the published Hacker News ranking algorithm. We will create a front page report that reflects this ranking order. The algorithm works in a way that nothing stays on the front page for too long, so a story’s score will eventually drop to zero over time (the gravity effect). Since our posts are spread out over a year and as older posts will always have a lower (or zero) ranking, we will distort the data so all our stories have the same date and and consider only the times in the ranking score calculation. This will give all our posts a fair chance of landing the front page.  Our front page report is published at 11:59pm. [Here's some additional information on the ranking.](http://www.righto.com/2013/11/how-hacker-news-ranking-really-works.html)
+
+### The plan
```
Comment (zeibura) [inline COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md:221]: Caps

File: COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md [hunk 4]
```diff
@@ -203,3 +203,55 @@ I hope that by taking this challenge, you will be able to learn something new!
...
+
+ [] We will then use `DFSORT` to sort the output dataset on ranking score, highest to lowest and display the posts as a simple report mimicking the front page. 
+ 
+ Let's get started!
+1. Take a look and familialize yourself with the dataset on z/OS: `ZOS.PUBLIC.HACKER.NEWS`. This is a CSV file that serves as input to your COBOL program. The file was created by downloading [this Kaggle dataset](https://www.kaggle.com/hacker-news/hacker-news-posts), removing the lengthy `URL` column that is of no relevance to us and uploading it to z/OS. You can directly reference this DS in your JCL. Please avoid making a copy as it is fairly large with around 300,000 records.
+
+ ![](Images/hacker-img2.png)
+  
+2.	Create your COBOL program in `<userid>.CBL` in the Code4z editor – This program will :
```
Comment (zeibura) [inline COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md:231]: I might change "in the Code4z editor" to "in VS Code with the Code4z extensions installed and enabled". Code4z doesn't really have an editor per se.

**Comment by jellypuno** [comment]
Hi @MikeBauerCA @zeibura , Can you please review this PR again? I would really like this to be merged. Thank you so much!

File: COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md [hunk 5]
```diff
@@ -203,3 +203,56 @@ I hope that by taking this challenge, you will be able to learn something new!
...
+
+### The Plan
+ [] There are different creative ways of accomplishing this but here’s our plan: We will have a COBOL program that reads the input CSV file and retrieves only the ***Mainframe/COBOL*** stories. It then calculates the ranking score for the stories by factoring in the time they were posted and the number of votes they received. Each of the records is then written to an output dataset along with the ranking score.
```
Comment (MikeBauerCA) [inline COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md:223]: Should [] be removed?

File: COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md [hunk 6]
```diff
@@ -203,3 +203,56 @@ I hope that by taking this challenge, you will be able to learn something new!
...
+
+### The Plan
+ [] There are different creative ways of accomplishing this but here’s our plan: We will have a COBOL program that reads the input CSV file and retrieves only the ***Mainframe/COBOL*** stories. It then calculates the ranking score for the stories by factoring in the time they were posted and the number of votes they received. Each of the records is then written to an output dataset along with the ranking score. 
+
+ [] We will then use `DFSORT` to sort the output dataset on ranking score, highest to lowest and display the posts as a simple report mimicking the front page.
```
Comment (MikeBauerCA) [inline COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md:225]: Should [] be removed?

**Comment by MikeBauerCA** [comment]
@raven300 I'll mark as approved. Thanks for making the change. The DCO signoff is failing because it wants you to put raven300 as your name in the signoff. You will need to head to your local branch on your machine to resolve the DCO signoff issue. Be sure to pull first.

**Comment by raven300** [comment]
Done. Thank you!

**Comment by MikeBauerCA** [comment]
@zeibura once you are satisfied with your requested changes, please approve and merge :)

**Pull request merged**